hashicorp/dev-portal
# feat: add button to download api-docs json

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-leah-featapi-docs-json-download-hashicorp.vercel.app/hcp/api-docs/vault-secrets) 🔎
- [Asana task](https://app.asana.com/0/1207339219333499/1207348593326770/f) 🎟️

## 🗒️ What


This PR adds a download button to the api-docs view. This way users can use the api spec in a tool like [redoc](https://redocly.github.io/redoc/) to generate helpful documentation for each method. I added a link to the page that will download this spec under the name `hcp.swagger.json`.

## 📸 Design Screenshots

<img width="535" alt="Screenshot 2024-10-29 at 4 07 04 PM" src="https://github.com/user-attachments/assets/f7a8252e-b566-4aa9-bfd7-5a4e4a192aff">



## 🧪 Testing



1. Go to [preview](https://dev-portal-git-leah-featapi-docs-json-download-hashicorp.vercel.app/hcp/api-docs/vault-secrets)
2. Click the 'Download spec' button
3. Take the downloaded file and upload it to [redoc](https://redocly.github.io/redoc/)
4. Verify that the output matches the dev-portal docs
